flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -300,17 +311,21 @@ public int hashCode() {
 										previousEvent,
 										previousTimestamp,
 										oldVersion);
+									validPreviousState = true;
 								}
 
-								// a new computation state is referring to the shared entry
-								sharedBuffer.lock(newState, event, timestamp);
+								if (validPreviousState) {
+									// a new computation state is referring to the shared entry
+									sharedBuffer.lock(newState, event, timestamp);
+
+									resultingComputationStates.add(new ComputationState<T>(
+										newState,
+										event,
+										timestamp,
+										newComputationStateVersion,
+										startTimestamp));
```

Comment:
Ideally we would filtered out all computation states which referred to pruned computations. Then we would not have to make this case distinction here.